### ISSUE TITLE
[Checkbox] aria-required should be added for accessibility

### ISSUE THREAD

**Issue by mathias-falkenberg** [mui/base-ui#1775]
# Bug report

## Current behavior

When applying the `required` prop to `Checkbox.Root`, only a `data-required` attribute is added. Assistive tech will not announce the Checkbox as required.

## Expected behavior

> When form controls are created using non-semantic elements, such as a `<div>` with a [role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles) of [checkbox](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/checkbox_role), the aria-required attribute should be included, with a value of true, to indicate to assistive technologies that user input is required on the element for the form to be submittable. The aria-required attribute can be used with HTML form elements; it is not limited to elements that have an ARIA role assigned.

[https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-required#description](url)

## Base UI version

1.0.0-alpha.7

## Which browser are you using?

Chrome/Edge

## Which OS are you using?

Windows

## Which assistive tech are you using (if applicable)?

Microsoft Narrator
